script.js

```diff
@@ -1,12 +1,235 @@
+    let episodeNumber = addZero(episodeList[i].number);
+    if (episodeList[i].image !== null && episodeList[i].summary !== null) {
+      aTag.href = `${episodeList[i].url}`;
+      aTag.target = "_blank";
+      h1Element.innerText = `${episodeList[i].name} - S${seasonNumber}E${episodeNumber}`;
+      imageElement.src = `${episodeList[i].image.medium}`;
+      pElement.innerHTML = `${truncate(episodeList[i].summary)}`;
+
+      //appending elements
+      aTag.appendChild(h1Element);
+      sectionElement.appendChild(aTag);
+      sectionElement.appendChild(imageElement);
+      sectionElement.appendChild(pElement);
+      mainElement.appendChild(sectionElement);
+      rootElem.appendChild(mainElement);
+    }
+  }
+}
+
+//search box
+function searchBox(kindOfShow) {
+  inputElement.addEventListener("keyup", function (event) {
+    let inputLetter = event.target.value.toLowerCase();
+    let episodes = document.getElementsByClassName("showSection");
+    let numOfResult = 0;
+    for (let i = 0; i < episodes.length; i++) {
+      const currentEpisode = episodes[i].textContent.toLowerCase();
+      if (currentEpisode.match(inputLetter)) {
+        // Display matching episodes
+        episodes[i].style.display = "block";
+        numOfResult += 1;
+      } else {
+        // Hide non-matching episodes
+        episodes[i].style.display = "none";
+      }
+    }
+    allEpisodes.innerText = `${episodes.length} ${kindOfShow}`;
+    searchedEpisode.innerText = `${numOfResult}`;
+  });
+}
+
+// Drop-down option menu
+function dropDown(episodes) {
+  let firstOption = document.createElement("option");
+  firstOption.innerHTML = "Select all episodes";
+  selectEpisodeElement.appendChild(firstOption);
+  for (let i = 0; i < episodes.length; i++) {
+    let optionElement = document.createElement("option");
+    if (episodes[i].image !== null && episodes[i].summary !== null) {
+      optionElement.value = `<section>
+    <a href = ${episodes[i].url}><h3>S${episodes[i].season}E${
+        episodes[i].number
+      } - ${episodes[i].name}</h3></a>
+    <img src = ${episodes[i].image.medium}>
+    <p>${truncate(episodes[i].summary)}</p>
+    </section>`;
+
+      let seasonNumber = addZero(episodes[i].season);
+      let episodeNumber = addZero(episodes[i].number);
+      optionElement.innerHTML = `S${seasonNumber}E${episodeNumber} - ${episodes[i].name}`;
+      selectEpisodeElement.appendChild(optionElement);
+    }
+    selectEpisodeElement.addEventListener("change", () => {
+      if (selectEpisodeElement.selectedIndex === 0) {
+        selectedShow();
+      } else {
+        reset();
+        mainElement.innerHTML = selectEpisodeElement.value;
+        rootElem.appendChild(mainElement);
+      }
+    });
+  }
+}
+
+//select show
+let selectShow = function () {
+  mainElement.innerHTML = "";
+  const showList = getAllShows();
+  showList.sort((a, b) => (a.name > b.name ? 1 : -1));
+  var showArr = [];
```

Comment:
Can you think of a way to do this with an array of objects rather than two separate arrays? I.e. a variable that looks like

```js
[
    {show: "show 1", id: 0 },
    {show: "another show", id: 1},
    // ...
];
```